internal/store/builder.go

```diff
@@ -546,7 +561,10 @@ func (b *Builder) buildCustomResourceStores(resourceName string,
 
 	var familyHeaders []string
 	if b.hasResources(resourceName, expectedType) {
-		familyHeaders = generator.ExtractMetricFamilyHeaders(metricFamilies)
+		if !compareToLastMetricFamily(metricFamilies) {
```

Comment:
is it safe to assume that it's the consecutive metrics family that you need to compare with?

Reply:
This is only to prevent multiple family creations under the same name from violating the OpenMetrics standards by cleanly grouping them under the same family name. I've added a comment stating the same. The condition should be safe in that use case.